src/app/resources/resources.service.ts

```diff
@@ -128,4 +131,38 @@ export class ResourcesService {
     return mediaTypes.find((type) => mimeType.indexOf(type) > -1) || 'other';
   }
 
+  sendResourceNotification(resource) {
+    const createdDate = resource.createdDate;
+    const currentUser = this.userService.get();
+    const resourceDocs = [];
+    const resourceNotification = forkJoin([
+      this.usersService.getAllUsers(),
+      this.couchService.findAll('notifications', findDocuments({ link: 'resources', type: 'newResource', status: 'unread' }))
+    ]).pipe(
+      switchMap(([ users, notifications ]: [ any[], any[] ]) => {
+        users.filter(user => {
+          return (currentUser.name !== user.name &&
+            user.name !== 'satellite' &&
+            notifications.every(notification => notification.user !== user._id));
+        }).map(user => resourceDocs.push(this.addResourceDoc(user, createdDate)));
```

Comment:
`Array.map` and `Array.filter` return a modified array.  You should use that value rather than performing a side effect to build a different array.